workflow-ui/container-android/src/main/java/com/squareup/workflow1/ui/modal/ModalContainer.kt

```diff
@@ -75,6 +101,12 @@ public abstract class ModalContainer<ModalRenderingT : Any> @JvmOverloads constr
               dialogView,
               findParentLifecycle = { parentLifecycleOwner?.lifecycle }
             )
+            // Ensure that each dialog has its own SavedStateRegistry so views in each dialog layer
+            // don't clash with other layers.
+            ref.stateRegistryOwner = KeyedStateRegistryOwner.installAsSavedStateRegistryOwnerOn(
+              view = dialogView,
+              key = Compatible.keyFor(newScreen, i.toString())
```

Comment:
Can this be built into `createStateRegistryKeyForContainer`? Add an optional `index: Int? = null` parameter, e.g.